tracer/src/Datadog.Trace/RemoteConfigurationManagement/Transport/RemoteConfigurationApi.cs

```diff
@@ -62,15 +63,15 @@ public async Task<GetRcmResponse> GetConfigs(GetRcmRequest request)
                 return null;
             }
 
+            var content = await apiResponse.ReadAsStringAsync().ConfigureAwait(false);
+
             if (apiResponse.StatusCode is not (>= 200 and <= 299))
             {
-                var content = await apiResponse.ReadAsStringAsync().ConfigureAwait(false);
                 Log.Warning<int, string>("Failed to receive remote configurations {StatusCode} and message: {ResponseContent}", apiResponse.StatusCode, content);
-
                 return null;
             }
 
-            return await apiResponse.ReadAsTypeAsync<GetRcmResponse>().ConfigureAwait(false);
+            return JsonConvert.DeserializeObject<GetRcmResponse>(content);
```

Comment:
If you're going to deserialize straight away here, I would use the more efficient JSON deserialization method to deserialize directly from the stream

Reply:
Agree, this should have been a temporary change to provide a log to a colleague. Reverted in 5a20c53da.